src/manifolds/ProbabilitySimplex.jl

```diff
@@ -0,0 +1,371 @@
+injectivity_radius(M::ProbabilitySimplex) = 0
+injectivity_radius(M::ProbabilitySimplex, ::SoftmaxRetraction) = 0
+injectivity_radius(M::ProbabilitySimplex, ::ExponentialRetraction) = 0
+eval(quote
+    @invoke_maker 1 Manifold injectivity_radius(
+        M::ProbabilitySimplex,
+        rm::AbstractRetractionMethod,
+    )
+end)
+
+@doc raw"""
+    inner(M::ProbabilitySimplex,p,X,Y)
+
+Compute the inner product of two tangent vectors `X`, `Y` from the tangent space $T_pΔ^n$ at
+`p`. The formula reads
+````math
+g_p(X,Y) = \sum_{i=1}^{n+1}\frac{X_iY_i}{p}
+````
+"""
+function inner(::ProbabilitySimplex, p, X, Y)
+    d = zero(Base.promote_eltype(p, X, Y))
+    @inbounds for i in eachindex(p, X, Y)
+        d += X[i] * Y[i] / p[i]
+    end
+    return d
+end
+
+@doc raw"""
+    inverse_retract(M::ProbabilitySimplex, p, q, ::SoftmaxInverseRetraction)
+
+Compute a first order approximation by projection. The formula reads
+````math
+\operatorname{retr}^{-1}_p q = \bigl( I - \frac{1}{n}\mathbb{1}\mathbb{1}^\mathrm{T}\bigr)(\log(q)-\log(p))
+````
+where $\mathbb{1}$ is the column vector containing ones and $\log$ is applied elementwise.
+"""
+inverse_retract(::ProbabilitySimplex, ::Any, ::Any, ::SoftmaxInverseRetraction)
+
+function inverse_retract!(
+    ::ProbabilitySimplex{n},
+    X,
+    p,
+    q,
+    ::SoftmaxInverseRetraction,
+) where {n}
+    X .= log.(q) .- log.(p)
+    meanlogdiff = mean(X)
+    X .-= meanlogdiff
+    return X
+end
+
+
+@doc raw"""
+    log(M::ProbabilitySimplex, p, q)
+
+Compute the logarithmic map of `p` and `q` on the [`ProbabilitySimplex`](@ref) `M`.
+
+````math
+\log_pq = \frac{d_{Δ^n}(p,q)}{\sqrt{1-⟨\sqrt{p},\sqrt{q}⟩}}(\sqrt{pq} - ⟨\sqrt{p},\sqrt{q}⟩p),
+````
+
+where $pq$ and $\sqrt{p}$ is meant elementwise.
+"""
+log(::ProbabilitySimplex, ::Any...)
+
+function log!(M::ProbabilitySimplex, X, p, q)
+    if p ≈ q
+        fill!(X, 0)
+    else
+        z = sqrt.(p .* q)
+        s = sum(z)
+        X .= 2 * acos(s) / sqrt(1 - s^2) .* (z .- s .* p)
+    end
+    return X
+end
+
+@doc raw"""
+    manifold_dimension(M::ProbabilitySimplex{n})
+
+Returns the manifold dimension of the probability simplex in $ℝ^{n+1}$, i.e.
+````math
+    \dim_{Δ^n} = n.
+````
+"""
+manifold_dimension(::ProbabilitySimplex{n}) where {n} = n
+
+"""
+    mean(
+        S::ProbabilitySimplex,
+        x::AbstractVector,
+        [w::AbstractWeights,]
+        method = GeodesicInterpolationWithinRadius(a);
+        kwargs...,
+    )
+
+Compute the Riemannian [`mean`](@ref mean(M::Manifold, args...)) of `x` using
+[`GeodesicInterpolationWithinRadius`](@ref), where the minimum of the injectivity radii
+of the points in `x` is used as the value `a`.
+"""
+mean(::ProbabilitySimplex, ::Any...)
+
+function mean!(M::ProbabilitySimplex, p, x::AbstractVector, w::AbstractVector; kwargs...)
+    a = minimum(injectivity_radius.(Ref(M), x))
+    return mean!(M, p, x, w, GeodesicInterpolationWithinRadius(a); kwargs...)
```

Comment:
We don't need the radius check here, a noted in my earlier comment.

```suggestion
    return mean!(M, p, x, w, GeodesicInterpolation(); kwargs...)
```

Reply:
Oh, I also had thought I did that earlier, I had some strange behaviour of my sourcetree where I seem to have lost a few lines before committing them. Will change that and update the docstring.